crates/exex/exex/src/manager.rs

```diff
@@ -139,6 +143,74 @@ impl ExExHandle {
     }
 }
 
+/// A stream of [`ExExNotification`]s returned by [`ExExNotificationsSubscriber::subscribe`] or
+/// [`ExExNotificationsSubscriber::subscribe_with_head`].
+pub type ExExNotifications<'a> =
+    Either<ExExNotificationsWithoutHead<'a>, ExExNotificationsWithHead<'a>>;
+
+/// A subscriber for [`ExExNotifications`].
+#[derive(Debug)]
+pub struct ExExNotificationsSubscriber {
+    notifications: Receiver<ExExNotification>,
+}
+
+impl ExExNotificationsSubscriber {
+    /// Creates a new [`ExExNotificationsSubscriber`].
+    pub const fn new(notifications: Receiver<ExExNotification>) -> Self {
+        Self { notifications }
+    }
+
+    /// Subscribe to notifications.
+    pub fn subscribe(&mut self) -> ExExNotifications<'_> {
+        ExExNotifications::Left(ExExNotificationsWithoutHead(&mut self.notifications))
```

Comment:
why are we wrapping this here?

Reply:
`mpsc::Receiver` doesn't implement `Stream` by default. https://docs.rs/tokio-stream/0.1.15/tokio_stream/wrappers/struct.ReceiverStream.html exists for this, but I didn't want to import a new dep just for that

Reply:
using either here is a bit weird, I'd prefer a custom type over this